apps/studio/components/interfaces/Advisors/CreateRuleSheet.tsx

```diff
@@ -0,0 +1,272 @@
+
+  const routeCategory = router.pathname.split('/').pop()
+  const organization = useSelectedOrganization()
+  const { data: members = [] } = useOrganizationMembersQuery({ slug: organization?.slug })
+
+  const { mutate: createRule, isLoading: isCreating } = useLintRuleCreateMutation({
+    onSuccess: (_, vars) => {
+      const ruleLint = vars.exception.lint_name
+      const ruleLintMeta = lintInfoMap.find((x) => x.name === ruleLint)
+      toast.success(`Successfully created new rule for ${ruleLintMeta?.title}`)
+
+      if (ruleLintMeta) {
+        if (!!routeCategory && routeCategory !== ruleLintMeta.category) {
+          router.push(
+            `/project/${projectRef}/advisors/rules/${ruleLintMeta.category}?lint=${ruleLintMeta.name}`
+          )
+        } else {
+          setExpandedLint(ruleLintMeta?.name)
+        }
+      }
+      onOpenChange(false)
+    },
+  })
+
+  const formId = 'create-lint-rule-form'
+  const form = useForm<z.infer<typeof FormSchema>>({
+    mode: 'onBlur',
+    reValidateMode: 'onChange',
+    resolver: zodResolver(FormSchema),
+    defaultValues,
+  })
+
+  const { lint_name, assigned_to, is_disabled } = form.watch()
+
+  const onSubmit: SubmitHandler<z.infer<typeof FormSchema>> = async (values) => {
+    if (!projectRef) return console.error('Project ref is required')
+
+    createRule({
+      projectRef,
+      exception: {
+        ...values,
+        lint_category: undefined,
+        lint_name: values.lint_name as LintName,
+        assigned_to: values.assigned_to === 'all' ? undefined : values.assigned_to,
+      },
+    })
+  }
+
+  useEffect(() => {
+    if (open) form.reset({ ...defaultValues, lint_name: lint?.name })
+  }, [open])
+
+  return (
+    <Sheet open={open} onOpenChange={onOpenChange}>
+      <SheetContent className="flex flex-col gap-0">
+        <SheetHeader className="shrink-0 flex items-center gap-4">
+          <SheetTitle>Create a rule on Advisor lints</SheetTitle>
+        </SheetHeader>
+        <SheetSection className="overflow-auto flex-grow px-0">
+          <Form_Shadcn_ {...form}>
+            <form
+              id={formId}
+              className="flex flex-col gap-y-4"
+              onSubmit={form.handleSubmit(onSubmit)}
+            >
+              <FormField_Shadcn_
+                key="lint_name"
+                name="lint_name"
+                control={form.control}
+                render={({ field }) => (
+                  <FormItemLayout layout="vertical" className="px-5" label="Lint name">
+                    <FormControl_Shadcn_>
+                      <Select_Shadcn_ onValueChange={field.onChange} defaultValue={field.value}>
+                        <SelectTrigger_Shadcn_ className="col-span-8">
+                          <SelectValue_Shadcn_ placeholder="Select a lint name" />
+                        </SelectTrigger_Shadcn_>
+                        <SelectContent_Shadcn_>
+                          {lintInfoMap.map((x) => (
```

Comment:
nit: I can see both performance and security lints in this list... kinda strange having this at all tbh 😅
 
<img width="605" alt="Screenshot 2025-03-25 at 15 20 43" src="https://github.com/user-attachments/assets/82802043-478c-470f-8840-2ea392510d6e" />

Reply:
yeah agreed - i was wondering whether to hide this field tbh, think i will hahaha given that you also find this weird

Reply:
will remove as such, and also beefed up the description under "What this rule means" a bit more
![image](https://github.com/user-attachments/assets/b63bb7f7-9656-4833-8ba8-36eb8b50f668)